tools/generate_pv_yaml_rest.sh

```diff
@@ -0,0 +1,393 @@
+  exit 2
+fi
+
+if [ -z "${VOLNAME}" ]; then
+  VOLNAME=${VOLPATH%/}
+  VOLNAME=${VOLNAME##*/}
+  VOLNAME="pv-${FSNAME}-${VOLNAME}"
+  VOLNAME=${VOLNAME,,}
+  if [[ ${#VOLNAME} -ge 254 ]]; then
+    echo "ERROR: Specify name for pv using option --pvname"
+    exit 2
+  fi
+
+  if ! [[ "${VOLNAME}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+    echo "ERROR: Specify name for pv using option --pvname"
+    exit 2
+  fi
+fi
+
+if ! [[ "${VOLNAME}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+  echo "ERROR: Invalid pv name specified. pv name must satisfy DNS-1123 label requirement."
+  exit 2
+fi
+
+[[ -z "${ACCESSMODE}" ]] && ACCESSMODE="ReadWriteMany"
+
+if ! [[ "$ACCESSMODE" == "ReadWriteMany" || "$ACCESSMODE" == "ReadWriteOnce" ]]; then
+  echo "ERROR: Invalid access mode specified. Valid accessmode are ReadWriteMany and ReadWriteOnce."
+  exit 2
+fi
+
+STORAGECLASS=""
+if ! [[ -z "${CLASS}" ]]; then
+  if ! [[ "${CLASS}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+    echo "ERROR: Invalid storageClass name specified. storageClass name must satisfy DNS-1123 label requirement."
+    exit 2
+  fi
+  STORAGECLASS="storageClassName: ${CLASS}"
+fi
+
+# Check if this is spectrum scale node
+#if [[ ! -f /usr/lpp/mmfs/bin/mmlscluster ]]; then
+#  echo "ERROR: Spectrum Scale cli's are not present on this node"
+#  exit 2
+#fi
+
+echo >${ERROROUT}
+
+# Authentication and route validation
+response=$(curl -kv -u "${USERNAME}":"${PASSWORD}" -X GET \
+  --header 'accept:application/json' \
+  "https://${URL}:443/scalemgmt/v2/cluster" \
+  2>&1 | grep -i 'HTTP/1.1 ' | awk '{print $3}'| sed -e 's/^[ \t]*//')
+
+if [[  ${response} == 401 ]]; then
+  echo "ERROR: Unauthorized. Incorrect username or password."
+  exit 2
+elif [[  -z ${response} ]]; then
+  echo "ERROR: Could not resolve host ${URL}."
+  exit 2
+fi
+
+# Get the Spectrum Scale cluster ID
+clusterID=$(curl -k -u "${USERNAME}":"${PASSWORD}" -X GET \
+  --header 'accept:application/json' \
+  "https://${URL}:443/scalemgmt/v2/cluster" \
+  2>${ERROROUT} | python3 -c "import sys, json; print(json.load(sys.stdin)['cluster']['clusterSummary']['clusterId'])" 2>>${ERROROUT})
+if [[ $? -ne 0 ]] || [[ -z "$clusterID" ]]; then
+  echo "ERROR: Failed to get the Spectrum Scale cluster ID."
+  #cat ${ERROROUT}
+  exit 2
+fi
+
+# Get the Fileystem ID
+fileSystemID=$(curl -k -u "${USERNAME}":"${PASSWORD}" -X GET \
+  --header 'accept:application/json' \
+  "https://${URL}:443/scalemgmt/v2/filesystems/${FSNAME}" \
+  2>${ERROROUT} | python3 -c "import sys, json; print(json.load(sys.stdin)['filesystems'][0]['uuid'])" 2>>${ERROROUT})
+if [[ $? -ne 0 ]] || [[ -z "$fileSystemID" ]]; then
+  echo "ERROR: Failed to get the Fileystem ID of ${FSNAME}"
+  #cat ${ERROROUT}
+  exit 2
+fi
+
+# TODO : Add check for kubernetes lable limit for value of VolumeHandle
+
+echo "FSETNAME=${FSETNAME}"
```

Comment:
is this for user's information? otherwise we can comment this echo statement ?

Reply:
Commented.